src/storage.cc

```diff
@@ -7,25 +7,18 @@ struct fse_event {
   UInt64 id;
   UInt32 flags;
   CFStringRef path;
+  
+  fse_event(CFStringRef eventPath, UInt32 eventFlag, UInt64 eventId) {
+    this->path = eventPath;
+    this->flags = eventFlag;
+    this->id = eventId;
+    if (this->path)
+      CFRetain(this->path);
+  }
+  
+  ~fse_event(){
+    if (this->path)
+      CFRelease(this->path);
+  }
```

Comment:
Now that it's juggling references, you should probably make the class non-copyable:

```c++
fse_event(const fse_event&);
void operator=(const fse_event&);
```